lock-impl/src/main/java/com/palantir/lock/logger/SanitizedLockRequestProgress.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.logger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.lock.impl.LockServiceStateDebugger;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableSanitizedLockRequestProgress.class)
+@JsonDeserialize(as = ImmutableSanitizedLockRequestProgress.class)
+public interface SanitizedLockRequestProgress {
+    String UNKNOWN_NEXT_LOCK_MESSAGE = "<unknown; this may have happened if the request has been serviced,"
+            + " but we have not cleared it from the map>";
+
+    List<SimpleLockRequest> getRequests();
+    String getNextLock();
+    int getNumLocksAcquired();
+    int getTotalNumLocks();
+
+    static SanitizedLockRequestProgress create(LockServiceStateDebugger.LockRequestProgress progress,
+            LockDescriptorMapper descriptorMapper,
+            String clientId) {
+        return ImmutableSanitizedLockRequestProgress.builder()
+                .totalNumLocks(progress.getTotalNumLocks())
+                .numLocksAcquired(progress.getNumLocksAcquired())
+                .nextLock(progress.getNextLock().map(descriptorMapper::getDescriptorMapping)
+                        .orElse(UNKNOWN_NEXT_LOCK_MESSAGE))
+                .requests(
+                        StreamSupport.stream(progress.getRequest().getLockDescriptors().entries().spliterator(), false)
+                                .map(descriptor -> SimpleLockRequest.of(
```

Comment:
Need to also backport SimpleLockRequest :(

Reply:
Yep, done.